cmake/app/boilerplate.cmake

```diff
@@ -43,6 +48,19 @@ include($ENV{ZEPHYR_BASE}/cmake/extensions.cmake)
 
 find_package(PythonInterp 3.4)
 
+# Determine if we are using MSYS.
+#
+# We don't use project() because it would take some time to rewrite
+# the build scripts to be compatible with everything project() does.
+execute_process(
+  COMMAND
+  uname
+  OUTPUT_VARIABLE uname_output
+  )
+if(uname_output MATCHES "^MSYS")
```

Comment:
does it need to be at the beginning of the string?

Reply:
I don't know. It was on my machine at least. I haven't looked up the docs.

Do you think we should look up what it looks like across platforms?

Reply:
there's only one platform, but I would simply match "MSYS"

Reply:
Sure, why not.